feat(shed): inspect miner fee payment at each deadline cron

rvagg opened the pull request:
Extension to https://github.com/filecoin-project/lotus/pull/12980 that looks for fee payment.

<details>
<summary>Output from butterflynet f01009</summary>

Current head is somewhere past 19126, we're within deadline 35 here so the latest cron is deadline 34, before that is historical, after that is even more historical.

```
Deadline 0:
	Sector 92 daily fee: <legacy>
	Sector 93 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 1:
	Sector 94 daily fee: <legacy>
	Sector 95 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 2:
	Sector 5 daily fee: <legacy>
	Sector 96 daily fee: <legacy>
	Sector 97 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1610612736 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 3:
	Sector 6 daily fee: <legacy>
	Sector 7 daily fee: <legacy>
	Sector 98 daily fee: <legacy>
	Sector 99 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 2147483648 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 4:
	Sector 8 daily fee: <legacy>
	Sector 9 daily fee: <legacy>
	Sector 104 daily fee: 621283336099
	Sector 105 daily fee: 621283336099
	✓ Deadline daily fee: 1242566672198, power: 11811160064 (sector fee sum: 1242566672198, expiration fee deduction sum: 1242566672198)
	Inspecting last deadline cron @ 17326:
		Trace involving t01009: t01012->t01009[5]:0;
		Trace involving t01009: t00->t03[2]:0 t03->t04[5]:0 t04->t02[3]:0 t04->t01009[12]:0 t01009->t099[0]:1242566672198 t01009->t04[6]:0 t01009->t04[4]:0 t04->t02[4]:0 t03->t05[9]:0;
		Estimated deadline daily reward: 17732241441% of fee (fee not capped)
		✓ Burn value: 1242566672198
		✓ Balance Δ: -1242566672198, LockedFunds Δ: -1242566672198, FeeDebt: 0, 
Deadline 5:
	Sector 10 daily fee: <legacy>
	Sector 11 daily fee: <legacy>
	Sector 106 daily fee: 62128536649
	Sector 107 daily fee: 62128536649
	✓ Deadline daily fee: 124257073298, power: 2147483648 (sector fee sum: 124257073298, expiration fee deduction sum: 124257073298)
	Inspecting last deadline cron @ 17386:
		Trace involving t01009: t00->t03[2]:0 t03->t04[5]:0 t04->t02[3]:0 t04->t01009[12]:0 t01009->t099[0]:124257073298 t01009->t04[6]:0 t01009->t04[4]:0 t04->t02[4]:0 t03->t05[9]:0;
		Estimated deadline daily reward: 4579917468% of fee (fee not capped)
		✓ Burn value: 124257073298
		✓ Balance Δ: -124257073298, LockedFunds Δ: -124257073298, FeeDebt: 0, 
Deadline 6:
	Sector 12 daily fee: <legacy>
	Sector 13 daily fee: <legacy>
	Sector 108 daily fee: 621303861554
	Sector 109 daily fee: 621303861554
	✓ Deadline daily fee: 1242607723108, power: 11811160064 (sector fee sum: 1242607723108, expiration fee deduction sum: 1242607723108)
	Inspecting last deadline cron @ 17446:
		Trace involving t01009: t00->t03[2]:0 t03->t04[5]:0 t04->t02[3]:0 t04->t01009[12]:0 t01009->t04[4]:0 t04->t02[4]:0 t03->t05[9]:0;
		Estimated deadline daily reward: 0% of fee (fee capped @ 0)
		✓ Burn value: <burn not found>
		✓ Balance Δ: 0, LockedFunds Δ: 0, FeeDebt: 0, 
Deadline 7:
	Sector 14 daily fee: <legacy>
	Sector 15 daily fee: 62294989087
	✓ Deadline daily fee: 62294989087, power: 1073741824 (sector fee sum: 62294989087, expiration fee deduction sum: 62294989087)
	Inspecting last deadline cron @ 17506:
		Trace involving t01009: t00->t03[2]:0 t03->t04[5]:0 t04->t02[3]:0 t04->t01009[12]:0 t01009->t04[4]:0 t04->t02[4]:0 t03->t05[9]:0;
		Estimated deadline daily reward: 0% of fee (fee capped @ 0)
		✓ Burn value: <burn not found>
		✓ Balance Δ: 0, LockedFunds Δ: 0, FeeDebt: 0, 
Deadline 8:
	Sector 16 daily fee: 62294989087
	Sector 17 daily fee: 62294989087
	✓ Deadline daily fee: 124589978174, power: 1073741824 (sector fee sum: 124589978174, expiration fee deduction sum: 124589978174)
	Inspecting last deadline cron @ 17566:
		Trace involving t01009: t00->t03[2]:0 t03->t04[5]:0 t04->t02[3]:0 t04->t01009[12]:0 t01009->t099[0]:124589978174 t01009->t04[6]:0 t01009->t04[4]:0 t04->t02[4]:0 t03->t05[9]:0;
		Estimated deadline daily reward: 86859801804% of fee (fee not capped)
		✓ Burn value: 124589978174
		✓ Balance Δ: -124589978174, LockedFunds Δ: -124589978174, FeeDebt: 0, 
Deadline 9:
	Sector 18 daily fee: 62294989087
	Sector 19 daily fee: <legacy>
	✓ Deadline daily fee: 62294989087, power: 1073741824 (sector fee sum: 62294989087, expiration fee deduction sum: 62294989087)
	Inspecting last deadline cron @ 17626:
		Trace involving t01009: t00->t03[2]:0 t03->t04[5]:0 t04->t02[3]:0 t04->t01009[12]:0 t01009->t099[0]:62294989087 t01009->t04[6]:0 t01009->t04[4]:0 t04->t02[4]:0 t03->t05[9]:0;
		Estimated deadline daily reward: 77176415602% of fee (fee not capped)
		✓ Burn value: 62294989087
		✓ Balance Δ: -62294989087, LockedFunds Δ: -62294989087, FeeDebt: 0, 
Deadline 10:
	Sector 20 daily fee: <legacy>
	Sector 21 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 11:
	Sector 22 daily fee: <legacy>
	Sector 23 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 12:
	Sector 24 daily fee: <legacy>
	Sector 25 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 13:
	Sector 26 daily fee: <legacy>
	Sector 27 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 14:
	Sector 28 daily fee: <legacy>
	Sector 29 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 15:
	Sector 30 daily fee: <legacy>
	Sector 31 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 16:
	Sector 32 daily fee: 62125213707
	Sector 33 daily fee: <legacy>
	✓ Deadline daily fee: 62125213707, power: 1073741824 (sector fee sum: 62125213707, expiration fee deduction sum: 62125213707)
	Inspecting last deadline cron @ 18046:
		Trace involving t01009: t00->t03[2]:0 t03->t04[5]:0 t04->t02[3]:0 t04->t01009[12]:0 t01009->t099[0]:62125213707 t01009->t04[6]:0 t01009->t04[4]:0 t04->t02[4]:0 t03->t05[9]:0;
		Estimated deadline daily reward: 123052222200% of fee (fee not capped)
		✓ Burn value: 62125213707
		✓ Balance Δ: -62125213707, LockedFunds Δ: -62125213707, FeeDebt: 0, 
Deadline 17:
	Sector 34 daily fee: <legacy>
	Sector 35 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 18:
	Sector 36 daily fee: <legacy>
	Sector 37 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 19:
	Sector 38 daily fee: <legacy>
	Sector 39 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 20:
	Sector 40 daily fee: <legacy>
	Sector 41 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 21:
	Sector 42 daily fee: 62124685079
	Sector 43 daily fee: <legacy>
	✓ Deadline daily fee: 62124685079, power: 1073741824 (sector fee sum: 62124685079, expiration fee deduction sum: 62124685079)
	Inspecting last deadline cron @ 18347:
		Trace involving t01009: t00->t03[2]:0 t03->t04[5]:0 t04->t02[3]:0 t04->t01009[12]:0 t01009->t099[0]:62124685079 t01009->t04[6]:0 t01009->t04[4]:0 t04->t02[4]:0 t03->t05[9]:0;
		Estimated deadline daily reward: 63001687850% of fee (fee not capped)
		✓ Burn value: 62124685079
		✓ Balance Δ: -62124685079, LockedFunds Δ: -62124685079, FeeDebt: 0, 
Deadline 22:
	Sector 44 daily fee: <legacy>
	Sector 45 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 23:
	Sector 46 daily fee: <legacy>
	Sector 47 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 24:
	Sector 48 daily fee: <legacy>
	Sector 49 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 25:
	Sector 50 daily fee: 621240779382
	Sector 51 daily fee: <legacy>
	✓ Deadline daily fee: 621240779382, power: 5905580032 (sector fee sum: 621240779382, expiration fee deduction sum: 621240779382)
	Inspecting last deadline cron @ 18586:
		Trace involving t01009: t00->t03[2]:0 t03->t04[5]:0 t04->t02[3]:0 t04->t01009[12]:0 t01009->t099[0]:621240779382 t01009->t04[6]:0 t01009->t04[4]:0 t04->t02[4]:0 t03->t05[9]:0;
		Estimated deadline daily reward: 27474422201% of fee (fee not capped)
		✓ Burn value: 621240779382
		✓ Balance Δ: -621240779382, LockedFunds Δ: -621240779382, FeeDebt: 0, 
Deadline 26:
	Sector 52 daily fee: <legacy>
	Sector 53 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 27:
	Sector 54 daily fee: <legacy>
	Sector 55 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 28:
	Sector 56 daily fee: <legacy>
	Sector 57 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 29:
	Sector 58 daily fee: <legacy>
	Sector 59 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 30:
	Sector 60 daily fee: <legacy>
	Sector 61 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 31:
	Sector 100 daily fee: 62124787714
	Sector 101 daily fee: 62124787714
	✓ Deadline daily fee: 124249575428, power: 1073741824 (sector fee sum: 124249575428, expiration fee deduction sum: 124249575428)
	Inspecting last deadline cron @ 18946:
		Trace involving t01009: t00->t03[2]:0 t03->t04[5]:0 t04->t02[3]:0 t04->t01009[12]:0 t01009->t099[0]:124249575428 t01009->t04[6]:0 t01009->t04[4]:0 t04->t02[4]:0 t03->t05[9]:0;
		Estimated deadline daily reward: 20001265688% of fee (fee not capped)
		✓ Burn value: 124249575428
		✓ Balance Δ: -124249575428, LockedFunds Δ: -124249575428, FeeDebt: 0, 
Deadline 32:
	Sector 102 daily fee: 62124787714
	Sector 103 daily fee: 62124787714
	✓ Deadline daily fee: 124249575428, power: 1073741824 (sector fee sum: 124249575428, expiration fee deduction sum: 124249575428)
	Inspecting last deadline cron @ 19006:
		Trace involving t01009: t00->t03[2]:0 t03->t04[5]:0 t04->t02[3]:0 t04->t01009[12]:0 t01009->t099[0]:124249575428 t01009->t04[6]:0 t01009->t04[4]:0 t04->t02[4]:0 t03->t05[9]:0;
		Estimated deadline daily reward: 19437415788% of fee (fee not capped)
		✓ Burn value: 124249575428
		✓ Balance Δ: -124249575428, LockedFunds Δ: -124249575428, FeeDebt: 0, 
Deadline 33:
	Sector 62 daily fee: <legacy>
	Sector 63 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 34:
	Sector 64 daily fee: <legacy>
	Sector 65 daily fee: 621235285058
	✓ Deadline daily fee: 621235285058, power: 5905580032 (sector fee sum: 621235285058, expiration fee deduction sum: 621235285058)
	Inspecting last deadline cron @ 19126:
		Trace involving t01009: t00->t03[2]:0 t03->t04[5]:0 t04->t02[3]:0 t04->t01009[12]:0 t01009->t099[0]:621235285058 t01009->t04[6]:0 t01009->t04[4]:0 t04->t02[4]:0 t03->t05[9]:0;
		Estimated deadline daily reward: 20294678192% of fee (fee not capped)
		✓ Burn value: 621235285058
		✓ Balance Δ: -621235285058, LockedFunds Δ: -621235285058, FeeDebt: 0, 
Deadline 35:
	Sector 66 daily fee: <legacy>
	Sector 67 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 36:
	Sector 68 daily fee: <legacy>
	Sector 69 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 37:
	Sector 70 daily fee: <legacy>
	Sector 71 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 38:
	Sector 72 daily fee: <legacy>
	Sector 73 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 39:
	Sector 74 daily fee: <legacy>
	Sector 75 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 40:
	Sector 76 daily fee: <legacy>
	Sector 77 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 41:
	Sector 78 daily fee: <legacy>
	Sector 79 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 42:
	Sector 80 daily fee: <legacy>
	Sector 81 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 43:
	Sector 82 daily fee: <legacy>
	Sector 83 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 44:
	Sector 84 daily fee: <legacy>
	Sector 85 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 45:
	Sector 86 daily fee: <legacy>
	Sector 87 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 46:
	Sector 88 daily fee: <legacy>
	Sector 89 daily fee: <legacy>
	✓ Deadline daily fee: 0, power: 1073741824 (sector fee sum: 0, expiration fee deduction sum: 0)
Deadline 47:
	Sector 90 daily fee: 621271647512
	Sector 91 daily fee: 621277250527
	✓ Deadline daily fee: 1242548898039, power: 10737418240 (sector fee sum: 1242548898039, expiration fee deduction sum: 1242548898039)
	Inspecting last deadline cron @ 17026:
		Trace involving t01009: t00->t03[2]:0 t03->t04[5]:0 t04->t02[3]:0 t04->t01009[12]:0 t01009->t04[3]:0 t01009->t04[4]:0 t04->t02[4]:0 t03->t05[9]:0;
		Estimated deadline daily reward: 0% of fee (fee capped @ 0)
		✓ Burn value: <burn not found>
		✓ Balance Δ: 0, LockedFunds Δ: 0, FeeDebt: 0, 
Total miner daily fee: 5716385437073 / 0.000005716385437073 FIL (✓ no discrepancies found)
```
</details>

ZenGround0 commented on cmd/lotus-shed/miner-fees.go:
```diff
@@ -148,9 +165,172 @@ var minerFeesCmd = &cli.Command{
+					return xerrors.Errorf("computing tipset at deadline end epoch: %w", err)
+				}
+
+				_, _ = fmt.Fprintf(cctx.App.Writer, "\tInspecting last deadline cron @ %d:\n", dlEndTs.Height())
```
handle error maybe?  What do we have to lose

rvagg replied:
There's 14 of them in this file, so we lose 14 x 2 = 28 more lines in this file of boilerplate that won't be triggered, I could write these as `fmt.Printf` for the same effect and then get to pretend there's never an error

Alternatively we wait for this so we can just put a `?` on the end of the line: https://github.com/golang/go/issues/71460

Also:

```
$ git grep =\ fmt.Fprintf.*ctx.App | wc
     70     758    7863
$ git grep =\ fmt.Fprintf.*ctx.App | grep -v '\s*_, _' | wc
      0       0       0
```

ZenGround0 reviewed (approved):
LGTM

rvagg commented:
Made some changes here:

* Put it under `lotus-shed miner` as `fees` now
* Added a `--proving-periods` that defaults to `1` so you can run through that many days of historical deadlines to check (this was useful just now to check the faulty period that James had on butterflynet)
* Fixed a bug in `lotus-shed miner locked-vested`
